### ISSUE TITLE
Orphan StringName when using match statement in GDScript

### ISSUE THREAD

**Issue by endragor** [godotengine/godot#15627]
**Godot version:**
3.0 3a05d974e0fb925869fe0fdfdbde9da8ee6a2757

**OS/device including version:**
macOS 10.12.6, likely every other platform

**Issue description:**
Using match statement in GDScript results in:

```
Orphan StringName: #match_value
StringName: 1 unclaimed string names at exit.
```

**Steps to reproduce:**
Run [this project](https://github.com/godotengine/godot/files/1626494/project.zip) with `godot -v --path .`, close it, see the logs.

**Comment by akien-mga** [comment]
CC @karroffel

**Comment by vnen** [comment]
This is already fixed (probably by #21411)
